beangulp/importers/csvbase.py

```diff
@@ -291,35 +301,39 @@ def extract(self, filepath, existing):
         offset = int(self.skiplines) + bool(self.names) + 1
 
         for lineno, row in enumerate(self.read(filepath), offset):
-            # Skip empty lines.
-            if not row:
-                continue
-
-            tag = getattr(row, 'tag', None)
-            tags = {tag} if tag else EMPTY
-
-            link = getattr(row, 'link', None)
-            links = {link} if link else EMPTY
-
-            # This looks like an exercise in defensive programming
-            # gone too far, but we do not want to depend on any field
-            # being defined other than the essential ones.
-            flag = getattr(row, 'flag', self.flag)
-            payee = getattr(row, 'payee', None)
-            account = getattr(row, 'account', default_account)
-            currency = getattr(row, 'currency', self.currency)
-            units = data.Amount(row.amount, currency)
-
-            # Create a transaction.
-            txn = data.Transaction(self.metadata(filepath, lineno, row),
-                                   row.date, flag, payee, row.narration, tags, links, [
-                                       data.Posting(account, units, None, None, None, None),
-                                   ])
-
-            # Apply user processing to the transaction.
-            txn = self.finalize(txn, row)
-            if txn is None:
-                continue
+            try:
+                # Skip empty lines.
+                if not row:
+                    continue
+
+                tag = getattr(row, 'tag', None)
+                tags = {tag} if tag else EMPTY
+
+                link = getattr(row, 'link', None)
+                links = {link} if link else EMPTY
+
+                # This looks like an exercise in defensive programming
+                # gone too far, but we do not want to depend on any field
+                # being defined other than the essential ones.
+                flag = getattr(row, 'flag', self.flag)
+                payee = getattr(row, 'payee', None)
+                account = getattr(row, 'account', default_account)
+                currency = getattr(row, 'currency', self.currency)
+                units = data.Amount(row.amount, currency)
+
+                # Create a transaction.
+                txn = data.Transaction(self.metadata(filepath, lineno, row),
+                                    row.date, flag, payee, row.narration, tags, links, [
+                                        data.Posting(account, units, None, None, None, None),
+                                    ])
+
+                # Apply user processing to the transaction.
+                txn = self.finalize(txn, row)
+                if txn is None:
+                    continue
+            except Exception:
+                print(f'Error processing file {filepath}, line {lineno}: {row!r}', file=sys.stderr)
```

Comment:
Printing to standard error is not the way to do this.

Reply:
Can you suggest a better way? My intention was to make it easier to debug which row causes parsing error when dealing with all kinds of not-so-standard CSV files.

Currently it's just a generic "importer error" without reavling the culprit row. It would be better to include this information in the traceback, but I don't know how.

Reply:
The proper way to do this is to chain exceptions. I'll implement this.